lib/agent0/agent0/hyperdrive/policies/lpandarb.py

```diff
@@ -104,14 +107,25 @@ def calc_reserves_to_hit_target_rate(
         # we modify bonds by only half of bonds_needed, knowing that an amount of equal
         # value will move shares in the other direction, toward our desired ratio.
         # this guess is very bad when slippage is high, so we check how bad, then scale accordingly.
-        bonds_needed = (target_bonds - pool_state.pool_info.bond_reserves) / FixedPoint(2)
-        shares_to_pool, shares_to_gov = calc_shares_needed_for_bonds(bonds_needed, pool_state, interface)
-        # save bad first guess to a temporary variable
-        temp_pool_state = apply_step(deepcopy(pool_state), bonds_needed, shares_to_pool, shares_to_gov)
-        predicted_rate = interface.calc_fixed_rate(temp_pool_state)
+        # to avoid negative share reserves, we increase the divisor until they are no longer negative.
+        divisor = FixedPoint(2)
+        bonds_needed = FixedPoint(0)
+        avoid_negative_share_reserves = False
+        while avoid_negative_share_reserves is False:
+            bonds_needed = (target_bonds - pool_state.pool_info.bond_reserves) / divisor
+            try:
+                shares_to_pool, shares_to_gov = calc_shares_needed_for_bonds(bonds_needed, pool_state, interface)
+                # save bad first guess to a temporary variable
+                temp_pool_state = apply_step(deepcopy(pool_state), bonds_needed, shares_to_pool, shares_to_gov)
+                predicted_rate = interface.calc_fixed_rate(temp_pool_state)
+                avoid_negative_share_reserves = temp_pool_state.pool_info.share_reserves >= 0
+            except:  # pylint: disable=bare-except
```

Comment:
if you're going to broad except (which you shouldn't) then please add a comment indicating what sort of issue you're expecting to have to avoid. And honestly it'd be best if you figured out what type of error that issue causes and catch that explicitly.

Reply:
removed